core/src/test/scala/org/apache/spark/deploy/master/MasterSuite.scala

```diff
@@ -327,22 +328,25 @@ class MasterSuite extends SparkFunSuite
     val masterUrl = s"http://localhost:${localCluster.masterWebUIPort}"
     try {
       eventually(timeout(5.seconds), interval(100.milliseconds)) {
-        val json = Source.fromURL(s"$masterUrl/json").getLines().mkString("\n")
+        val json = Utils
+          .tryWithResource(Source.fromURL(s"$masterUrl/json"))(_.mkString)
```

Comment:
I know you changed per the review comment above but let's keep the contents. At least it wouldn't keep the original files as are anymore, and it's strictly a different issue from what you're fixing here.

Reply:
ok